fix from upstream for --no-export-dynamic on PIE files

http://bugs.gentoo.org/311081

2010-03-21  H.J. Lu  <hongjiu.lu@intel.com>

	PR ld/11413
	* elflink.c (_bfd_elf_add_default_symbol): Check !executable
	instead of shared.

===================================================================
RCS file: /cvs/src/src/bfd/elflink.c,v
retrieving revision 1.370
retrieving revision 1.371
diff -u -r1.370 -r1.371
--- src/bfd/elflink.c	2010/03/05 19:48:55	1.370
+++ src/bfd/elflink.c	2010/03/21 23:26:33	1.371
@@ -1715,7 +1715,7 @@
 	{
 	  if (! dynamic)
 	    {
-	      if (info->shared
+	      if (! info->executable
 		  || hi->ref_dynamic)
 		*dynsym = TRUE;
 	    }
@@ -1784,7 +1784,7 @@
 	    {
 	      if (! dynamic)
 		{
-		  if (info->shared
+		  if (! info->executable
 		      || hi->ref_dynamic)
 		    *dynsym = TRUE;
 		}
